crates/swc_ecma_transforms_typescript/tests/fixture/issue-3073/4/output.js

```diff
@@ -2,20 +2,20 @@ var A;
 (function(A) {
     const a = 1;
     const b = 2;
-    ({ a: A.A  } = {
+    const { a: A1 } = { a: A.A } = {
```

Comment:
Is this correct?

Reply:
I think it's OK.
The exported name(`A`) is correct.
The local ident is rename to `A1` because of hygiene/resolved pass.

Reply:
I have tested the following input:

```TypeScript
module A {
    const a = 1;
    export const { a: A } = { a };

    console.log(A);
}
```

Output:
```TypeScript
var A;
(function(A) {
    const a = 1;
    const { a: A1 } = { a: A.A } = {
        a
    };
    console.log(A1);
})(A || (A = {}));

```

Reply:
I mean, it seems like there's duplicate bindings.

Reply:
No, there is only one binding.

```
const { a: A1 } = { a: A.A } = {
            |          ^^^---------- assign `a` to `A.A`
            |-- local binding
```

Reply:
https://github.com/swc-project/swc/blob/06fec63ceb9ff09fb540fa08e5bf9f7816ddbd13/crates/swc_ecma_transforms_typescript/src/transform.rs#L594-L611

Reply:
Thanks!